lib/mix/lib/mix/sync/lock.ex

```diff
@@ -312,20 +318,36 @@ defmodule Mix.Sync.Lock do
 
     lock_path = Path.join(path, "lock_0")
 
-    # We linked to lock_N successfully, so port_path should exist
-    File.rename!(port_path, lock_path)
+    # We linked to lock_N successfully, so port_path should exist.
+    # On Windows, renaming to an existing destination returns :eexist.
+    # In that case, another process won the race; we signal a retry.
```

Comment:
There should be no races here, only one process should be able to obtain `lock_N` and go into take-over.

We need to figure out in which circumstances `File.rename/2` fails and find a workaround for that. It is expected that the file may be open temporarily, as another Elixir process reads its content while locking; in that case we could wait a bit and retry the rename.